runtime/compiler/optimizer/EstimateCodeSize.cpp

```diff
@@ -44,13 +45,21 @@ TR_EstimateCodeSize::get(TR_InlinerBase * inliner, TR_InlinerTracer *tracer, int
    estimator->_isLeaf = false;
    estimator->_foundThrow = false;
    estimator->_hasExceptionHandlers = false;
-   estimator->_throwCount = 0;
    estimator->_mayHaveVirtualCallProfileInfo = false;
 
-   TR_CatchBlockProfileInfo * catchInfo = TR_CatchBlockProfileInfo::get(comp);
-   estimator->_aggressivelyInlineThrows = catchInfo && catchInfo->getCatchCounter()
-         >= TR_CatchBlockProfileInfo::EDOThreshold;
 
+   if (comp->getOption(TR_EnableOldEDO))
+      {
+      TR_CatchBlockProfileInfo * catchInfo = TR_CatchBlockProfileInfo::get(comp);
+      estimator->_aggressivelyInlineThrows = catchInfo && catchInfo->getCatchCounter() >= comp->getOptions()->getCatchBlockCounterThreshold();
+      }
+   else
+      {
+      TR::Recompilation *recomp = comp->getRecompilationInfo();
+      estimator->_aggressivelyInlineThrows = !comp->getOption(TR_DisableEDO) &&
+                                             recomp &&
+                                             recomp->getMethodInfo()->getCatchBlockCounter() >= comp->getOptions()->getCatchBlockCounterThreshold();
```

Comment:
It's hard to compare the thresholds in the two implementations. In the old implementation the JIT would profile catch blocks only if the optimizer decided to switch to a profiling compilation to profile some value of interest. So catch block profiling is a second class citizen. The profiling body runs until the desired number of samples for the value to be tracked is achieved, but this doesn't have much correlation with the catch block counter.

Reply:
Right, so given that, do you expect tuning of the default setting of the `catchBlockCounterThreshold` option will be needed, just as [you mentioned in the review for OMR pull request #6922](https://github.com/eclipse/omr/pull/6922#pullrequestreview-1334944135) that there would be tuning of the `edoRecompSizeThreshold` and `edoRecompSizeThresholdInStartupMode` options.  Or is 50 the right default value for `catchBlockCounterThreshold` with the new implementation?

Reply:
Yes, I will do some tuning for `catchBlockCounterThreshold`. I should have done that already but got distracted with some other higher priority items. We can postpone merging this PR until I am done tuning. Since I very rarely touch the optimizer, I wanted a review to make sure the implementation won't change drastically from what I have today. Thanks for your thorough review!